src/api/class-jw-player-api.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * WP Video Sync: JW Player API integration.
+ *
+ * @package wp-video-sync
+ */
+
+namespace Alley\WP\WP_Video_Sync\API;
+
+class JW_Player_API {
+
+	/**
+	 * The API URL.
+	 *
+	 * @var string
+	 */
+	public string $api_url = 'https://api.jwplayer.com/v2/sites';
+
+	/**
+	 * The API public key.
+	 *
+	 * @var string
+	 */
+	public string $api_key;
+
+	/**
+	 * The API v2 secret key.
+	 *
+	 * @var string
+	 */
+	public string $api_secret;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct( string $api_key, string $api_secret ) {
+		$this->api_key    = $api_key;
+		$this->api_secret = $api_secret;
+	}
+
+	/**
+	 * Set the user agent in the request headers for identification purposes.
+	 *
+	 * @return string
+	 */
+	public function user_agent(): string {
+		global $wp_version;
+
+		return 'WordPress/' . $wp_version . ' WPVideoSync/' . WP_VIDEO_SYNC_VERSION . ' PHP/' . phpversion();
+	}
+
+	/**
+	 * Generate the request URL.
+	 *
+	 * @param string $last_modified_date The date of the last modification to the last batch of videos.
+	 * @param int    $batch_size         The number of videos to fetch in each batch.
+	 *
+	 * @return string
+	 */
+	public function request_url( string $last_modified_date, int $batch_size ): string {
+		$request_url = $this->api_url . '/' . $this->api_key . '/media/';
+
+		return add_query_arg(
+			[
+				'q'           => 'last_modified:[' . $last_modified_date . ' TO *]',
+				'page'        => 1,
+				'sort'        => 'last_modified:asc',
+				'page_length' => $batch_size,
+			],
+			$request_url
+		);
+	}
+
+	/**
+	 * Generate the request arguments.
+	 *
+	 * @param string $type The type of request to make.
+	 *
+	 * @return array
+	 */
+	public function request_args( string $type = '' ): array {
+		$default_args = [
+			'user-agent' => $this->user_agent(),
+			'headers'    => [
+				'Authorization' => 'Bearer ' . $this->api_secret,
+				'Content-Type'  => 'application/json',
+			],
+		];
+
+		return 'vip' === $type
```

Comment:
Rather than having type be a string that could be `vip` either allow the user to specify the timeout value directly or let them specify an array of args for the remote get function